static/js/components/admin/MembersList.js

```diff
@@ -0,0 +1,69 @@
+// @flow
+import React from "react"
+import { Link } from "react-router-dom"
+
+import { profileURL } from "../../lib/url"
+import { MISSING_TEXT } from "../../lib/channels"
+
+import type { Channel, Member } from "../../flow/discussionTypes"
+
+type Props = {
+  channel: Channel,
+  editable: boolean,
+  members: Array<Member>,
+  usernameGetter: (member: Member) => string,
+  removeMember: (channel: Channel, username: string) => Promise<*>
+}
+
+export default class MembersList extends React.Component<Props> {
```

Comment:
Nit: could refactor this into a stateless component since react can optimize those a bit